### PR TITLE
feat: create tag group and tag item

### PR DESCRIPTION
**Pull request by mimoham24**
# Overview
`createTagItem` and `createTagGroup` mutations 
## What I've done
- create graphQL schema
- create the presentation layer
- create dummy infrastructure/memory layer (will create another PR implementing it)
- create use-case layer
- implement creating tag items and groups mutations

## What I haven't done
implementing `infrastructure/memory`
## How I tested
providing dummy data to the mutations

### CODE REVIEW

**Review by HideBa — APPROVED** [review]
LGTM

File: internal/usecase/interactor/tag.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err2 := tx.End(ctx); err == nil && err2 != nil {
+			err = err2
+		}
+	}()
+
+	if err := i.CanReadScene(ctx, inp.SceneID, operator); err != nil {
```
Comment (rot1024) [inline internal/usecase/interactor/tag.go:39]:
```suggestion
	if err := i.CanWriteScene(ctx, inp.SceneID, operator); err != nil {
```

File: internal/usecase/interactor/tag.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
```
Comment (rot1024) [inline internal/usecase/interactor/tag.go:77]:
```suggestion
	if err := i.CanWriteScene(ctx, inp.SceneID, operator); err != nil {
```

File: schema.graphql [hunk 3]
```diff
@@ -706,6 +706,33 @@ type MergedInfoboxField {
...
+  linkedDatasetSchema: DatasetSchema @goField(forceResolver: true)
+  linkedDataset: Dataset @goField(forceResolver: true)
+  linkedDatasetField: DatasetField @goField(forceResolver: true)
+}
+
+type TagGroup implements Tag {
+  id: ID!
+  sceneId: ID!
+  label: String!
+  tags: [ID]
```
Comment (rot1024) [inline schema.graphql:731]:
```suggestion
  tags: [ID!]!
```

File: schema.graphql [hunk 4]
```diff
@@ -1046,6 +1073,20 @@ input AddDatasetSchemaInput {
...
+  label: String!
+  linkedDatasetSchemaID: ID
+  linkedDatasetID: ID
+  linkedDatasetField: ID
+}
+
+input CreateTagGroupInput{
+  sceneId: ID!
+  label: String!
+  tags: [ID]
```
Comment (rot1024) [inline schema.graphql:1087]:
```suggestion
  tags: [ID!]!
```

File: schema.graphql [hunk 5]
```diff
@@ -1046,6 +1073,20 @@ input AddDatasetSchemaInput {
...
+  sceneId: ID!
+  label: String!
+  linkedDatasetSchemaID: ID
+  linkedDatasetID: ID
+  linkedDatasetField: ID
+}
+
+input CreateTagGroupInput{
+  sceneId: ID!
+  label: String!
```
Comment (rot1024) [inline schema.graphql:1086]:
```suggestion
  label: String
```

When the user does not want to update the name, the user won't set the label field

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: schema.graphql (hunk 5 above)
Comment (rot1024) [reply schema.graphql:1086]: Note: apply this suggestion only for UpdateTag*Input

**Review by rot1024 — APPROVED** [review]

**Pull request merged**